tests/test_envs.py

```diff
@@ -1,73 +1,98 @@
-from gymnasium.utils.env_checker import check_env
+import pickle
+
+import gymnasium as gym
+from gymnasium.utils.env_checker import check_env, data_equivalence
 
 import mo_gymnasium as mo_gym
 from mo_gymnasium import LinearReward
 
 
+def _test_pickle_env(env: gym.Env):
+    pickled_env = pickle.loads(pickle.dumps(env))
+
+    data_equivalence(env.reset(), pickled_env.reset())
+
+    action = env.action_space.sample()
+    data_equivalence(env.step(action), pickled_env.step(action))
+    env.close()
+    pickled_env.close()
+
+
 def test_deep_sea_treasure():
```

Comment:
You should look into `@pytest.mark.parameterize()`, this would simplify your testing